### DISCUSSION TITLE
proto test directory configuration

### DISCUSSION THREAD

**Discussion by Grantismo** [quarkusio/quarkus#43293] [Q&A]
Is it possible to configure a test directory for protos messages used exclusively in tests? 

I see in https://quarkus.io/guides/grpc-generation-reference that you can configure the proto directory via
`quarkus.grpc.codegen.proto-directory`, but I could not find any further configuration options. 

When writing generic proto processing code it's helpful to define proto messages that will only be used in test similar to https://github.com/entur/mapstruct-spi-protobuf/blob/7ffeba82db83d7ff36fe857038cb797f7371b62d/usage/src/main/proto/User.proto

[a comment by an automated account is not shown]

**Comment by cescoffier** [comment]
That's a good question. 
Did you try with `%test.quarkus.grpc.codegen.proto-directory`?. I wonder if it works.

If it does not. I think we need to revise the generation a bit to distinguish _main_ from _test_.


\CC @alesj

**Reply by Grantismo** [reply]
Unfortunately `%test.quarkus.grpc.codegen.proto-directory` does not appear to work.
I think ideally the configuration option would allow specifying multiple directories so I could keep test and prod proto directories separate but use both to generate protos for test if possible.

**Reply by cescoffier** [reply]
It's not surprising it does not work. Profiles are not applied during code generation.
@alesj can you check if it's possible to have different configuration for _main_ and _test_ codegen phase?
